app/src/main/java/com/concrete/magicthegathering/feature/detailcard/presentation/viewmodel/DetailCardViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.concrete.magicthegathering.feature.detailcard.presentation.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.viewModelScope
+import com.concrete.magicthegathering.core.base.BaseViewModel
+import com.concrete.magicthegathering.data.model.entity.favorite.CardFavorite
+import com.concrete.magicthegathering.feature.detailcard.repository.IDetailCardRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class DetailCardViewModel(private val repository: IDetailCardRepository) : BaseViewModel(), IDetailCardViewModel {
+    var getLiveDataCard: LiveData<CardFavorite?>? = null
+
+    override fun findByCard(multiverseid: Long) {
+        getLiveDataCard = repository.findByCard(multiverseid)
+    }
+
+    override fun insertOrRemoveCard(insertEnabled: Boolean, cardFavorite: CardFavorite) {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
```

Comment:
@CharlesMoreira1 O Room tem suporte nativo a coroutines, portanto, não é necessário usar o `withContext(Dispatcher.IO)` se o método estiver marcado com `suspend`. Isto se aplica tbm ao Retrofit que eu acabei esquecendo de comentar.

Reply:
Sim. Sobre o retrofit eu já sabia, tanto é que se vc for olhar o SetViewModel, ele faz direto na main.

Corrigido o do room*